sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java

```diff
@@ -485,6 +496,44 @@ protected boolean advanceImpl() throws IOException {
 
   /////////////////////////////////////////////////////////////////////////////
 
+  /** Implementation of {@link #ofProvider}. */
+  public static class OfValueProvider<T> extends PTransform<PBegin, PCollection<T>> {
+    private final ValueProvider<T> provider;
+    private final Coder<T> coder;
+
+    private OfValueProvider(ValueProvider<T> provider, Coder<T> coder) {
+      checkNotNull(provider, "provider");
+      checkNotNull(coder, "coder");
+      this.provider = provider;
```

Comment:
this.provider = checkNotNull(provider, "provider");
this.coder = checkNotNull(coder, "coder");